tango/step.py

```diff
@@ -242,6 +245,7 @@ def __init__(
         ] = None  # This is set only while the run() method runs.
         self._config = step_config
         self.step_resources = step_resources
+        self.workspace_metadata = workspace_metadata
```

Comment:
If you add a member named `workspace_metadata` to the class directly, and change this to `self.workspace_metadata.update(workspace_metadata)`, then you can specify metadata when you write the step class (which is useful for things like the artifact type, which will always be the same), and to the step constructor/configuration file, which is useful for things like machine limits (because bigger models need more resources than smaller ones).